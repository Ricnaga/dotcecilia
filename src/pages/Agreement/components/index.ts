export * from './AgreementCalculator/AgreementCalculator';
export * from './AgreementTableBody/AgreementTableBody';
