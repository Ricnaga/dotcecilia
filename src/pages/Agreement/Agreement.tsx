import { PageContainer, PaymentTable } from '@shared/components';
import { useInputFields } from '@shared/hooks/useInputFields';
import { useReactPrint } from '@shared/hooks/useReactPrint';
import {
  AgreementCalculator,
  AgreementCalculatorFields,
  AgreementTableBody,
} from './components';

export function Agreement() {
  const {
    data: { printRef },
    functions: { setPrint },
  } = useReactPrint();
  const {
    data: { values },
    functions: { onChangeValue },
  } = useInputFields<AgreementCalculatorFields>({
    name: '',
    salary: 0,
    startDate: new Date(),
    discount: 0,
    endDate: new Date(),
  });
  return (
    <PageContainer title="Acerto">
      <div className="grid grid-cols-3 gap-10">
        <div className="grid grid-cols-12 col-span-2" ref={printRef}>
          <PaymentTable
            headerType="ACERTO"
            name={values.name}
            startDate={values.startDate}
            endDate={values.endDate}
          >
            <AgreementTableBody values={values} />
          </PaymentTable>
        </div>
        <div>
          <AgreementCalculator
            values={values}
            onChange={onChangeValue}
            onPrint={setPrint}
          />
        </div>
      </div>
    </PageContainer>
  );
}
