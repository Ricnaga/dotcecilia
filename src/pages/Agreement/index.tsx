import { CeciliaPageHeader, CeciliaTable } from '@shared/components';
import { AgreementTableBody } from './components';

export function AgreementPage() {
  return (
    <CeciliaPageHeader title="Acerto">
      <div className="grid grid-cols-12 gap-10">
        <div className="grid grid-cols-12 col-span-7">
          <CeciliaTable headerType="ACERTO">
            <AgreementTableBody />
          </CeciliaTable>
        </div>
        <div className="col-span-5">
          <h1>oi</h1>
        </div>
      </div>
    </CeciliaPageHeader>
  );
}
