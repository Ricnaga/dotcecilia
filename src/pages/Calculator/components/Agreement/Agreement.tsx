import { InputCurrency } from '@shared/components/Input/InputCurrency/InputCurrency';
import { useState } from 'react';
import { InputNumber } from '@shared/components';
import { CalculatorTable, TBodyKey } from '../CalculatorTable/CalculatorTable';
import { CalculatorTd } from '../CalculatorTable/CalculatorTd';
import { CalculatorTr } from '../CalculatorTable/CalculatorTr';

export function Agreement() {
  const [value, setValue] = useState<
    Record<'salary' | 'workingMonths', number>
  >({
    salary: 0,
    workingMonths: 0,
  });

  const toBRL = (value: number) =>
    new Intl.NumberFormat('pt-br', {
      style: 'currency',
      currency: 'BRL',
    }).format(value);

  const dataSalary: TBodyKey = [
    {
      label: 'Férias',
      value: toBRL((value.salary / 12) * value.workingMonths),
    },
    {
      label: '13º',
      value: toBRL((value.salary / 12) * value.workingMonths),
    },
    {
      label: 'Férias - 1/3',
      value: toBRL(((value.salary / 12) * value.workingMonths) / 3),
    },
  ];

  return (
    <div className="grid grid-cols-2 gap-4">
      <div className="flex flex-col gap-2 justify-around">
        <InputCurrency
          placeholder="Salário"
          onChange={(value) =>
            setValue((state) => ({ ...state, salary: value }))
          }
        />
        <InputNumber
          placeholder="Meses trabalhados"
          onChange={(value) =>
            setValue((state) => ({ ...state, workingMonths: value }))
          }
        />
      </div>
      <CalculatorTable title="Salário">
        {dataSalary.map((data) => (
          <CalculatorTr key={data.label}>
            <CalculatorTd>{data.label}</CalculatorTd>
            <CalculatorTd>{data.value}</CalculatorTd>
          </CalculatorTr>
        ))}
      </CalculatorTable>
    </div>
  );
}
