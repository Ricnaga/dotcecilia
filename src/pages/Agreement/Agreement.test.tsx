import { render } from '@testing-library/react';
import { Agreement } from './Agreement';

describe('Page: Agreement', () => {
  it('should render correctly', () => {
    const { container } = render(<Agreement />);
    expect(container).toMatchSnapshot();
  });
});
