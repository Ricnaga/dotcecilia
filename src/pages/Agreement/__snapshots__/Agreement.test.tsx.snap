// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Agreement should render correctly 1`] = `
<div>
  <div>
    <p
      class="text-5xl font-bold mt-8 mb-6"
    >
      Acerto
    </p>
    <div
      class="grid grid-cols-3 gap-10"
    >
      <div
        class="grid grid-cols-12 col-span-2"
      >
        <table
          class="col-span-12 h-1 bg-slate-100 border border-slate-800 text-slate-900"
        >
          <thead
            class=""
          >
            <tr>
              <th
                colspan="6"
              >
                undefined
              </th>
              <th
                colspan="2"
              >
                RECIBO DE 
                ACERTO
              </th>
            </tr>
          </thead>
          <thead>
            <tr>
              <th
                class="text-right"
                colspan="6"
              >
                undefined
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                CNPJ: 
                undefined
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                <p
                  class="mt-4 border-y border-slate-800 bg-yellow-100"
                >
                  ACERTO: 
                  Ref. 22/09/2023 á 22/09/2023 -     
 (0) MESES
                </p>
              </th>
            </tr>
            <tr>
              <th
                colspan="8"
              >
                <p
                  class="my-4 border-y border-slate-800"
                >
                  Nome
                </p>
              </th>
            </tr>
          </thead>
          <tbody
            class="border-t border-slate-800"
          >
            <tr>
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                SALÁRIO:
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                Pagar
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                Desconto
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                13º
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                 0,00
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                FÉRIAS
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
              />
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-r border-slate-800 undefined"
                colspan="3"
              >
                1/3 - Férias
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              />
              <td
                class="border-r border-slate-800 text-center"
                colspan="2"
              >
                 0,00
              </td>
              <td
                class="border-r border-slate-800 text-center"
              />
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                colspan="6"
              >
                <p
                  class="mt-4 border-t border-slate-800"
                >
                  Sub.total:
                </p>
              </td>
              <td
                class="text-center"
              >
                <p
                  class="mt-4 border-t border-slate-800"
                >
                   0,00
                </p>
              </td>
              <td
                class="border-r border-slate-800 text-center"
              >
                <p
                  class="mt-4 border-t border-slate-800"
                >
                   0,00
                </p>
              </td>
            </tr>
            <tr
              class="border-t border-slate-800 undefined"
            >
              <td
                class="border-b border-slate-800"
                colspan="5"
              >
                <p
                  class="mb-4 border-b border-slate-800"
                >
                  Total á pagar:
                </p>
              </td>
              <td
                class="border-r border-slate-800 text-center"
                colspan="3"
              >
                <p
                  class="mb-4 border-b border-slate-800"
                >
                   0,00
                </p>
              </td>
            </tr>
          </tbody>
          <tfoot
            class="border-t border-slate-800"
          >
            <tr>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Assinatura:.......................................................
                </p>
              </td>
              <td
                colspan="4"
              >
                <p
                  class="my-8 text-center"
                >
                  Data:__/__/__
                </p>
              </td>
            </tr>
          </tfoot>
        </table>
      </div>
      <div>
        <div
          class="grid grid-cols-2 items-center"
        >
          <p
            class="text-4xl font-bold leading-9"
          >
            Cálculos
          </p>
          <button
            class="justify-self-end"
            data-variants="smoosh"
          >
            <span
              data-text="default"
            >
              Imprimir
            </span>
          </button>
        </div>
        <div
          class="grid grid-cols-2 gap-4 mt-8"
        >
          <div
            class="col-span-2"
          >
            <div>
              <div
                data-error="false"
                data-variants="input-wrapper"
              >
                <input
                  data-variants="input"
                  placeholder="Digite o nome"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <div>
              <div
                data-error="false"
                data-variants="input-wrapper"
              >
                <input
                  data-variants="input"
                  placeholder="salário"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <div>
              <div
                data-error="false"
                data-variants="input-wrapper"
              >
                <input
                  data-variants="input"
                  placeholder="Descontos (R$)"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <label
              class="text-lg ml-8 leading-6 font-medium"
              for="date"
            >
              Data inicial
              <div>
                <div
                  data-error="false"
                  data-variants="input-wrapper"
                >
                  <input
                    data-variants="input"
                    type="date"
                    value="2023-09-22"
                  />
                </div>
              </div>
            </label>
          </div>
          <div>
            <label
              class="text-lg ml-8 leading-6 font-medium"
              for="date"
            >
              Data Final
              <div>
                <div
                  data-error="false"
                  data-variants="input-wrapper"
                >
                  <input
                    data-variants="input"
                    type="date"
                    value="2023-09-22"
                  />
                </div>
              </div>
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
